### ISSUE TITLE
[BUG] {v,h}strip not showing when values are large and width/height is small on WebGL

### ISSUE THREAD

**Issue by hoxbro** [bokeh/bokeh#13478]
### Software versions

``` yaml
Python version      :  3.11.6 | packaged by conda-forge | (main, Oct  3 2023, 10:40:35) [GCC 12.3.0]
IPython version     :  8.16.1
Tornado version     :  6.3.3
Bokeh version       :  3.3.0
BokehJS static path :  /home/shh/miniconda3/envs/holoviz/lib/python3.11/site-packages/bokeh/server/static
node.js version     :  v20.8.1
npm version         :  10.1.0
Operating system    :  Linux-6.5.6-76060506-generic-x86_64-with-glibc2.35
```

### Expected behavior

WebGL should behave like canvas.

### Observed behavior

When using WebGL and having a large range of {v, h}strips with a small width/height, the strips are not always shown.

### Example code

```Python
import numpy as np
from bokeh.plotting import curdoc, figure, row

x0 = np.array(
    [109474856360, 109474856360, 109474858588, 109474859601, 109474860831]
).astype(np.float64)
x1 = x0 + 5

plot_canvas = figure(
    width=300, height=300, y_range=(0, 1), output_backend="canvas", title="Canvas"
)
plot_canvas.vstrip(x0=x0, x1=x1)

plot_webgl = figure(
    width=300, height=300, y_range=(0, 1), output_backend="webgl", title="WebGL"
)
plot_webgl.vstrip(x0=x0, x1=x1)

curdoc().add_root(row(plot_canvas, plot_webgl))
```

### Stack traceback or browser console output

Nothing. 

### Screenshots

https://github.com/bokeh/bokeh/assets/19758978/2250cd14-9c28-4c1d-889e-559f91545a0f

**Comment by ianthomas23** [comment]
Confirmed bug. This doesn't just affect `vstrip` and `hstrip`, it affects any WebGL marker (i.e. fixed-shape glyph).

In the example here the maths of the transform from data to screen units results in a glyph width of zero. Not a very small number, but `0.0`. For canvas the filled rectangle and edge line are drawn separately so even if a rectangle has zero width and isn't drawn, the line is still drawn. WebGL renders both fill and line at the same time, and if the width is zero we draw nothing. The fix is to avoid rendering if `width < 0.0` but allow it if `width == 0.0`as then you will see half the antialiased vertical line each side of the zero-width rectangle, which is essentially what canvas does.

**Comment by ianthomas23** [comment]
It is not quite that simple, PR #12362 specifically removed rendering of WebGL markers with zero size. 

The sensible solution is to not render markers with `width == height == 0.0` but do render them if only one of `width` and `height` is zero and the other is positive.
